### PR TITLE
More async tests

### PR DESCRIPTION
**Pull request by miiu96**
Added more async tests

### CODE REVIEW

File: integrationTests/vm/testIndexer.go [hunk 1]
```diff
@@ -168,7 +168,7 @@ func (ti *testIndexer) SaveTransaction(
 	ti.indexer.SaveBlock(blk, header, txsPool, nil, nil, nil)
 
 	// wait --> dataDispatcher needs to save the block
-	time.Sleep(5 * time.Millisecond)
+	time.Sleep(10 * time.Millisecond)
```
Comment (iulianpascalau) [inline integrationTests/vm/testIndexer.go:171]: can we make this time.Second as to be certain that won't fail when the TC agent is experiences high load

**Review by SebastianMarian — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

File: process/smartContract/process.go [hunk 2]
```diff
@@ -571,7 +571,8 @@ func (sc *scProcessor) computeTotalConsumedFeeAndDevRwd(
 		if !sc.isSelfShard(outAcc.Address) {
 			sentGas := uint64(0)
 			for _, outTransfer := range outAcc.OutputTransfers {
-				sentGas += outTransfer.GasLimit
+				sentGas, _ = core.SafeAddUint64(outTransfer.GasLimit, sentGas)
```
Comment (SebastianMarian) [inline process/smartContract/process.go:574]: sentGas, outTransfer.GasLimit would have been more natural :)

File: process/smartContract/process.go [hunk 3]
```diff
@@ -571,7 +571,8 @@ func (sc *scProcessor) computeTotalConsumedFeeAndDevRwd(
...
+				sentGas, _ = core.SafeAddUint64(outTransfer.GasLocked, sentGas)
```
Comment (SebastianMarian) [inline process/smartContract/process.go:575]: sentGas, outTransfer.GasLocked would have been more natural :)

**Review by SebastianMarian — APPROVED** [review]

File: process/smartContract/process.go [hunk 4]
```diff
@@ -585,7 +590,7 @@ func (sc *scProcessor) computeTotalConsumedFeeAndDevRwd(
 		log.LogIfError(err, "computeTotalConsumedFeeAndDevRwd", "computeGasLimit")
 	}
 
-	consumedGasWithoutBuiltin := consumedGas
+	consumedGasWithoutBuiltin, err := core.SafeSubUint64(consumedGas, accumulatedGasUsedForOtherShard)
```
Comment (SebastianMarian) [inline process/smartContract/process.go:593]: add LogIfError

**Review by sasurobert — DISMISSED** [review]

File: process/smartContract/process.go [hunk 5]
```diff
@@ -576,6 +577,10 @@ func (sc *scProcessor) computeTotalConsumedFeeAndDevRwd(
 			}
 			consumedGas, err = core.SafeSubUint64(consumedGas, sentGas)
 			log.LogIfError(err, "computeTotalConsumedFeeAndDevRwd", "outTransfer.GasLimit")
+			consumedGas, err = core.SafeAddUint64(consumedGas, outAcc.GasUsed)
+			log.LogIfError(err, "computeTotalConsumedFeeAndDevRwd", "outAcc.GasUsed")
```
Comment (SebastianMarian) [inline process/smartContract/process.go:581]: Both log errors L581 and L583 display the same things

File: process/smartContract/process.go (hunk 4 above)
Comment (sasurobert) [reply process/smartContract/process.go:593]: done

File: process/smartContract/process.go (hunk 5 above)
Comment (sasurobert) [reply process/smartContract/process.go:581]: done

File: process/smartContract/process.go (hunk 2 above)
Comment (sasurobert) [reply process/smartContract/process.go]: done

File: process/smartContract/process.go (hunk 3 above)
Comment (sasurobert) [reply process/smartContract/process.go]: done

**Review by sasurobert — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by stefanandy — APPROVED** [review]
System Test Passed

File: process/smartContract/process.go [hunk 6]
```diff
@@ -572,15 +572,15 @@ func (sc *scProcessor) computeTotalConsumedFeeAndDevRwd(
...
 			for _, outTransfer := range outAcc.OutputTransfers {
-				sentGas, _ = core.SafeAddUint64(outTransfer.GasLimit, sentGas)
-				sentGas, _ = core.SafeAddUint64(outTransfer.GasLocked, sentGas)
+				sentGas, _ = core.SafeAddUint64(sentGas, outTransfer.GasLimit)
+				sentGas, _ = core.SafeAddUint64(sentGas, outTransfer.GasLocked)
 			}
 			consumedGas, err = core.SafeSubUint64(consumedGas, sentGas)
 			log.LogIfError(err, "computeTotalConsumedFeeAndDevRwd", "outTransfer.GasLimit")
 			consumedGas, err = core.SafeAddUint64(consumedGas, outAcc.GasUsed)
 			log.LogIfError(err, "computeTotalConsumedFeeAndDevRwd", "outAcc.GasUsed")
```
Comment (SebastianMarian) [inline process/smartContract/process.go:581]: log.LogIfError(err, "computeTotalConsumedFeeAndDevRwd", "consumedGas", "outAcc.GasUsed")

File: process/smartContract/process.go [hunk 7]
```diff
@@ -572,15 +572,15 @@ func (sc *scProcessor) computeTotalConsumedFeeAndDevRwd(
 		if !sc.isSelfShard(outAcc.Address) {
 			sentGas := uint64(0)
 			for _, outTransfer := range outAcc.OutputTransfers {
-				sentGas, _ = core.SafeAddUint64(outTransfer.GasLimit, sentGas)
-				sentGas, _ = core.SafeAddUint64(outTransfer.GasLocked, sentGas)
+				sentGas, _ = core.SafeAddUint64(sentGas, outTransfer.GasLimit)
+				sentGas, _ = core.SafeAddUint64(sentGas, outTransfer.GasLocked)
 			}
 			consumedGas, err = core.SafeSubUint64(consumedGas, sentGas)
 			log.LogIfError(err, "computeTotalConsumedFeeAndDevRwd", "outTransfer.GasLimit")
```
Comment (SebastianMarian) [inline process/smartContract/process.go:579]: log.LogIfError(err, "computeTotalConsumedFeeAndDevRwd", "consumedGas", "sentGas")

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**